### PR TITLE
Port of Geocoding Components

### PR DESCRIPTION
**Pull request by theduckylittle**
Updated Geocoding and GeocodingResults  for redux.

There is also an update to the map reducer to support zooming to a given extent.  This is needed to support the ability to zoom to a given extent.

### CODE REVIEW

File: src/components/Geocoding.js [hunk 1]
```diff
@@ -10,88 +10,140 @@
...
-     )
-   }
- }
- // Maps state from store to props
- const mapStateToProps = (state, ownProps) => {
-   return {
-   }
- };
+      // kick off the query.
+      return Axios.get(this.state.search_url, {params}).then((response) => {
```
Comment (bartvde) [inline src/components/Geocoding.js]: I've asked Willie this as well, but wonder if we could go with simple fetch instead? Do we need axios?

File: src/components/GeocodingResults.js [hunk 2]
```diff
@@ -10,187 +10,144 @@
...
-    map.getView().setCenter(center);
-    map.getView().setZoom(this.props.zoom);
-    var source = this._layer.getSource();
-    source.clear();
-    var feature = new ol.Feature({
-      geometry: new ol.geom.Point(center)
+  render() {
+    // map all of the results to GeocodingResult items
+    const items = this.props.results.map((item, idx) => {
+      return (<GeocodingResult key={'result' + idx} zoomToExtent={this.props.zoomToExtent} result={item}/>);
```
Comment (bartvde) [inline src/components/GeocodingResults.js]: key can just be the idx? Doesn't really matter as long as it's unique in the set.

File: src/components/GeocodingResults.js [hunk 3]
```diff
@@ -10,187 +10,144 @@
...
+// from the state to render, instead using props.results.
+const mapStateToProps = (state, ownProps) => {
+  return {}
+};
+
+// Maps actions to props
+const mapDispatchToProps = (dispatch) => {
+  return {
+    // You can now say this.props.createBook
+    zoomToExtent: (extent) => dispatch(zoomToExtent(extent))
```
Comment (bartvde) [inline src/components/GeocodingResults.js]: comment seems irrelevant / inaccurate?

File: src/reducers/map.js [hunk 4]
```diff
@@ -1,4 +1,24 @@
 import {MAP} from '../actions/ActionTypes';
+
+import ol from 'openlayers';
+
+/** Take in the size of the map and extent and
+ *   return a new "view" containing center and zoom.
+ */
+
+function convertFromExtent(mapSize, extent, rotation) {
+  const view = new ol.View();
```
Comment (bartvde) [inline src/reducers/map.js]: theoretically this view could be different than the view the map is using? Wrt constraints etc? Projection etc.

Comment (theduckylittle) [inline src/reducers/map.js]: It could. I haven't setup a testing "rig" but I thought it would work like this:
1. The state changes to a view that is "out of bounds".
2. The map will go to it's "max" allowable bounds.
3. The "moveend" event will trigger with the new center/resolution based on the max allowable bounds.  This triggers the setView action which will have the state mirror the map.

File: src/actions/ActionTypes.js [hunk 5]
```diff
@@ -9,5 +9,7 @@ export const MAP = {
   SET_VIEW: 'MAP_SET_VIEW',
   ZOOM_IN: 'MAP_ZOOM_IN',
   ZOOM_OUT: 'MAP_ZOOM_OUT',
-  SET_ROTATION: 'MAP_SET_ROTATION'
+  SET_ROTATION: 'MAP_SET_ROTATION',
+  SET_SIZE: 'SET_SIZE',
+  EXTENT: 'EXTENT'
```
Comment (theduckylittle) [inline src/actions/ActionTypes.js]: Fixed. Used "FIT_..."

File: src/components/Geocoding.js [hunk 6]
```diff
@@ -10,88 +10,140 @@
...
-   _searchAddress(event) {
-     if (this.geocodeSearchText.input.value.length > 2) {
-       this.props.geocodingSearch(this.geocodeSearchText.input.value, event.currentTarget);
-     }
-   }
-   geocodeSearchText = '';
+    // seed the state as empty.
+    this.state = {
+      // use the nomantim search url if not given anything else.
+      search_url: props.searchUrl ? props.searchUrl : SEARCH_URL,
```
Comment (theduckylittle) [inline src/components/Geocoding.js]: converted to defaultProps instead.

File: src/components/Geocoding.js [hunk 7]
```diff
@@ -10,88 +10,140 @@
...
+      search_params: props.searchParams ? props.searchParams : SEARCH_PARAMS,
```
Comment (theduckylittle) [inline src/components/Geocoding.js]: same fix for this one

File: src/components/Geocoding.js (hunk 1 above)
Comment (theduckylittle) [inline src/components/Geocoding.js]: Dropped axios (included update of package.json)

File: src/components/GeocodingResults.js (hunk 2 above)
Comment (theduckylittle) [inline src/components/GeocodingResults.js]: Sorry, habits. The key=idx is :100: fine w/ me.

File: src/components/GeocodingResults.js (hunk 3 above)
Comment (theduckylittle) [inline src/components/GeocodingResults.js]: cruft. removed.

**Comment by theduckylittle** [comment]
This has now been updated to more closely reflect the new wire frames and components.

File: __tests__/components/Geocoding.test.js [hunk 8]
```diff
@@ -3,20 +3,32 @@
...
 import {assert} from 'chai';
-import intl from '../mock-i18n';
 import getMuiTheme from 'material-ui/styles/getMuiTheme';
 import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
-import {Geocoding} from '../../src/components/Geocoding';
+import {IntlProvider} from 'react-intl';
+
+import BoundlessSdk from '../../src/components/BoundlessSdk';
+
+import polyfills from '../polyfills'; // eslint-disable-line no-unused-vars
```
Comment (bartvde) [inline __tests__/components/Geocoding.test.js]: we can also change this one to ```import '../polyfills';```

File: __tests__/components/GeocodingResults.test.js [hunk 9]
```diff
@@ -1,16 +1,18 @@
...
-import {GeocodingResults} from '../../src/components/GeocodingResults';
-import {BoundlessSdk} from '../../src/components/BoundlessSdk';
-import * as geocodingActions from '../../src/actions/GeocodingActions';
 
-raf.polyfill();
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import {IntlProvider} from 'react-intl';
+
+import polyfills from '../polyfills'; // eslint-disable-line no-unused-vars
```
Comment (bartvde) [inline __tests__/components/GeocodingResults.test.js]: same here

**Comment by theduckylittle** [comment]
Fixed the polyfills inclusion as well.
